packages/components/pie-form-label/src/index.ts

```diff
@@ -38,6 +67,7 @@ export class PieFormLabel extends RtlMixin(LitElement) implements FormLabelProps
 
         return html`
             <label
+                @click=${this.handleClick}
```

Comment:
we would still need to ensure that this works like a native label from an A11y perspective - screenreaders should treat this like a native HTML label